app/modules/asset_groups/models.py

```diff
@@ -247,6 +247,39 @@ def commit(self):
     def has_filename(self, filename):
         return filename in self.config.get('assetReferences', [])
 
+    # Returns a percentage complete value 0-100
+    # function name is also the name of the value returned in the schema so has to be completion rather than the
+    # more intuitive get_completion
+    def completion(self):
+        # Design allows for these limits to be configured later, potentially this data could be project specific
+        stage_base_sizes = {
+            AssetGroupSightingStage.unknown: 0,
+            AssetGroupSightingStage.detection: 0,
+            AssetGroupSightingStage.curation: 10,
+            AssetGroupSightingStage.processed: 30,  # 40 for identification 20 for review
+            AssetGroupSightingStage.failed: 100,  # complete, even if failed
+        }
+        completion = stage_base_sizes[self.stage]
+
+        # some stages are either all or nothing, these just use the base sizes above.
+        # For those that have granularity we need to know the size range available and estimate how much has been done
+        if self.stage == AssetGroupSightingStage.detection:
+            if len(self.jobs) > 0:
+                size_range = (
+                    stage_base_sizes[AssetGroupSightingStage.curation]
+                    - stage_base_sizes[self.stage]
+                )
+                complete_jobs = [job for job in self.jobs if not job['active']]
+                completion += size_range * (len(complete_jobs) / len(self.jobs))
+        elif self.stage == AssetGroupSightingStage.processed:
+            assert len(self.asset_group_sightings) == 1
+            size_range = 100 - stage_base_sizes[self.stage]
+            sighting_completion = self.asset_group_sightings[0].get_completion()
```

Comment:
I am thinking that maybe asset_group_sightings is the wrong name. This is the sighting object associated with the asset_group_sighting. 
I could find a way to create a 1:1 sqlalchemy mapping so it has to be a 1:many and in this case, many = 1, hence the assert